text/0000-safecoin-alternative-implementation/0000-safecoin-alternative-implementation.md

```diff
@@ -0,0 +1,238 @@
+
+As with other network addressable entities, the `id` will represent the account's address on the network.  It will be managed by the close group of Vaults for that address, using a new persona: CoinManager.  The ID is also an asymmetric public signing key.
+
+A single client can create many `CoinAccount`s if desired, and there is no need to associate any of these with its main MAID account.  For example, a user may want to run multiple farmer Vaults and allocate a different `CoinAccount` for each to receive the farming rewards.
+
+When a client creates a new `CoinAccount` on the network, the request need only specify the `id`, since the initial balance will always be 0.  It is up to the client to create and manage the key pairs of all `CoinAccount`s it owns.  These could be stored on the network as part of its encrypted MAID account, or managed by a standalone wallet application.
+
+To pay for network resources (e.g. putting data) a `PaymentAccount` must exist where the account's ID is the same as the MAID's public key:
+
+```rust
+struct PaymentAccount {
+    details: CoinAccount,
+    app_rules: BTreeMap<PublicSigningKey, AppRules>,
+}
+
+enum AppRules {
+    Unrestricted,
+    SpendLimit {
+        remaining: Coin,
+    }
+}
+```
+
+The details of the `PaymentAccount` will be explained below at "[Paying for Network Resources](#paying-for-network-resources)".  Until we reach that point, just consider this as a wrapped `CoinAccount` and when `CoinAccount` is mentioned it means either of these types.
+
+
+
+### Farming
+
+Each section of the network will be responsible for a proportion of the total issuable safecoin equal to the proportion of the network address space it manages.  For instance, at the start when the network has a single section covering the full address space, that section will be responsible for all `2^32` safecoin.  When the section splits, each half will be responsible for `2^31` safecoin.
+
+This means that for a section with Prefix length `n`, it will be responsible for `2^(32-n)` safecoin.
+
+Each section's CoinManagers will maintain a record `farmed` (of type `Coin`) of the amount of safecoin farmed at that section.  Any changes to this total will come as a result of an action having passed through Parsec in order to ensure that all managers maintain an eventually consistent record.
+
+The section's `farmed` value will never be allowed to exceed the amount of coins for which that section is responsible.  Ideally no section should ever get close to "running out" of farmable coins; getting the farming rate algorithm correct should ensure that.  However, in the case that a section has farmed all of its coins, it will stop issuing any more until the `farmed` value reduces again.
+
+The actual algorithm defining the farming rate is somewhat orthogonal to this RFC and worthy of its own RFC, so the details will be omitted, but it would seem fairly critical that it ensures sections don't become fully farmed or fully unfarmed.
+
```

Comment:
possibly worth a simplified version here that states 
As resources drop below current data size + 50% or something we increase FR and at over data size + 50% we decrease farming rate.

Reply:
I was hoping to avoid any details here since I think it'd be a bit of a distraction from the main focus, and this RFC doesn't tie us to any particular algorithm.  What I really wanted to get across was that the chosen algorithm should ensure as far as possible that we don't run out of farmable coins or farmed coins.